plugin/pulpcore/plugin/changeset/iterator.py

```diff
@@ -48,11 +48,38 @@ def __iter__(self):
         while True:
             batch = tuple(itertools.islice(generator, 0, self.batch))
             if batch:
-                yield batch
+                yield Batch(batch, len(batch))
             else:
                 return
 
 
+class Batch(Iterable, Sized):
```

Comment:
Removing the `Batch` because it adds no value.  Not sure what I was thinking :-/